python/pyspark/sql/readwriter.py

```diff
@@ -773,6 +776,8 @@ def json(self, path, mode=None, compression=None, dateFormat=None, timestampForm
                                 formats follow the formats at ``java.text.SimpleDateFormat``.
                                 This applies to timestamp type. If None is set, it uses the
                                 default value, ``yyyy-MM-dd'T'HH:mm:ss.SSSXXX``.
+        :param encoding: specifies encoding (charset) of saved json files. If None is set,
+                        the default UTF-8 charset will be used.
```

Comment:
shall we mention that, if `encoding` is set, `lineSep` also need to be set when `multiLine` is false?

Reply:
Sorry, I didn't realized initially that the comment related to writing. For writing if `lineSep` is not set by user, it will be set to `\n`  in any case: https://github.com/MaxGekk/spark-1/blob/482b79969b9e0cc475e63b415051b32423facef4/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala#L124

Actually, the current implementation is more strict than it is needed. It requires to set `lineSep` explicitly in write if `multiLine` is `false` and `encoding` is different from `UTF-8`. I will change it.